src/main/scala/org/bblfsh/client/v2/JNode.scala

```diff
@@ -1,15 +1,60 @@
 package org.bblfsh.client.v2
 
+import scala.collection.mutable
+
 /** UAST nodes representation on the JVM side.
  *
  *  Mirrors https://godoc.org/github.com/bblfsh/sdk/uast/nodes
  */
-sealed abstract class JNode
-case class JString(s: String) extends JNode
-case class JDouble(num: Double) extends JNode
-case class JLong(num: Long) extends JNode
-case class JInt(num: BigInt) extends JNode
-case class JBool(value: Boolean) extends JNode
-case class JObject(obj: List[JField]) extends JNode
-case class JArray(arr: List[JNode]) extends JNode
+sealed abstract class JNode {
+  def children: Seq[JNode] = this match {
+    case JObject(l) => l map (_._2)
+    case JArray(l) => l
+    case _ => Seq()
+  }
+
+  def size: Int = this match {
+    case JObject(l) => l.size
+    case JArray(l) => l.size
+    case JString(l) => l.length
+    case JNothing => 0
+    case _ => 1
```

Comment:
What does 1 signify here? (And why 1 instead of, say, error?)

Reply:
It signifies a single element rather then a container one, and you are right, it should be 0 instead (same as in other clients).